csrc/scheduler/resize.cpp

```diff
@@ -0,0 +1,194 @@
+  return pointwise_utils::getReferenceTensor(fusion);
+}
+
+} // namespace
+
+bool ResizeScheduler::canScheduleCompileTime(Fusion* fusion) {
+  if (!ir_utils::hasOpsOfType<SliceOp, PadOp>(fusion)) {
+    scheduler_debug_utils::canScheduleRejectReason(
+        schedulerType(), "No resize op to schedule");
+    return false;
+  }
+
+  if (scheduler_utils::isResharding(fusion)) {
+    scheduler_debug_utils::canScheduleRejectReason(
+        schedulerType(), "Fusion is resharding.");
+    return false;
+  }
+
+  if (ir_utils::hasAnyReductionOps(fusion)) {
+    scheduler_debug_utils::canScheduleRejectReason(
+        schedulerType(), "No support for reduction ops");
+    return false;
+  }
+
+  if (registry_utils::hasNonUniqueBcast(fusion)) {
+    scheduler_debug_utils::canScheduleRejectReason(
+        schedulerType(),
+        "Broadcasting dimension might be broadcasting to multiple sizes.");
+    return false;
+  }
+
+  // For now, the resize scheduler is only allowed for a limited set
+  // of fusion patterns. The restrictions are planned to be
+  // incrementally relaxed.
+
+  IdModel id_model(fusion, /*build_graphs=*/false);
+  const auto& broadcast_graph = id_model.buildBroadcastGraph();
+
+  // For now, only a single resize op is allowed to exist.
+  auto resize_based_tensor_ops = ir_utils::getOpsOfType<SliceOp, PadOp>(fusion);
+  if (resize_based_tensor_ops.size() != 1) {
+    scheduler_debug_utils::canScheduleRejectReason(
+        schedulerType(), "Only a single resize op is allowed.");
+    return false;
+  }
+
+  // Slicing of or to a broadcast ID is not allowed yet.
+  for (auto tensor_op : resize_based_tensor_ops) {
+    TensorView* out_tv = tensor_op->output(0)->as<TensorView>();
+    for (auto logical_id : out_tv->getLogicalDomain()) {
+      Resize* resize = dynamic_cast<Resize*>(logical_id->definition());
+      if (resize == nullptr) {
+        continue;
+      }
+
+      if (resize->out()->isBroadcast()) {
+        scheduler_debug_utils::canScheduleRejectReason(
+            schedulerType(), "Resize to a broadcast ID is not allowed.");
+        return false;
+      }
+
+      // Need to check the broadcast group rather than just the input
+      // ID only. For example,
+      //
+      // t0: [i0]
+      // t1: [b1]
+      // t2 = t0 + t1
+      // t3 = slice(t2)
+      //
+      // Then, propagating the slice to its inputs would try to
+      // propagate the resize op to b1 as well, which would fail due
+      // to issue #3571
+      const auto& input_group = broadcast_graph.toGroup(resize->in());
+      if (std::any_of(
+              input_group->begin(), input_group->end(), [](Val* inp_val) {
+                return inp_val->as<IterDomain>()->isBroadcast();
+              })) {
+        scheduler_debug_utils::canScheduleRejectReason(
+            schedulerType(), "Resize of a broadcast ID is not allowed.");
+        return false;
+      }
+    }
+  }
+
+  // This doesn't work yet due to issue #3571
```

Comment:
for my own education. this means after #3571 is resolved, we'll remove the check in the forloop above and just rely on the check below. i.e. we no longer needed to build id_model for the canScheduleCompileTime check.

Reply:
The check will become much more complex, likely using IdModel